### PR TITLE
Add scp support

### PR DESCRIPTION
**Pull request by manicmaniac**
Add support for SCP.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,8 +8,8 @@ addons:
    repo_token: 4d2c1cec2a5ba5fd0cd09aa76d1bcb52854e12ace21660dbf65a36a59ba7a973
 
 rvm:
-  - 2.0.0
```
Comment (DanToml) [inline .travis.yml]: We still need to support ruby 2.0 in CocoaPods until Xcode 10 ships.

Comment (rito-m) [reply .travis.yml]: Thank you for your reviewing.
I'm going to manage to do so.

Comment (manicmaniac) [reply .travis.yml]: Thank you for your reviewing.
Perhaps I have to wait until this issue resolved... https://github.com/travis-ci/travis-ci/issues/9190 .

**Comment by manicmaniac** [comment]
@DanToml Fixed to work on ruby 2.0.0. Could you review this please?

File: lib/cocoapods-downloader/remote_file.rb [hunk 2]
```diff
@@ -0,0 +1,179 @@
+require 'zlib'
+require 'fileutils'
+require 'uri'
+
+module Pod
+  module Downloader
+    class RemoteFile < Base
```
Comment (segiddins) [inline lib/cocoapods-downloader/remote_file.rb]: this probably needs tests

Comment (manicmaniac) [reply lib/cocoapods-downloader/remote_file.rb]: moved some tests from http_specs.rb to remote_file_spec.rb.

**Comment by manicmaniac** [comment]
@segiddins Thanks for your review.
I think all problems are resolved. Please continue your review again.

**Comment by segiddins** [comment]
@DanToml ping for review?

**Comment by DanToml** [comment]
Sorry for taking a while, got lost in my gh notifications

**Pull request merged**